.github/workflows/ci.yaml

```diff
@@ -34,8 +36,12 @@ jobs:
         with:
           command: test
           args: --verbose
-      - name: Run clippy
-        uses: actions-rs/cargo@v1
-        with:
-          command: clippy
-          args: --verbose -- --deny warnings
+      - name: Run Clippy and redirect output to file
+        run: cargo clippy --verbose -- --deny warnings > clippy.log 2>&1 || true
+      - name: Create issue if clippy found warnings
+        run: |
+          if [ $(grep -c 'warning' clippy.log || true) != '0' ]; then
+            gh issue create --title "Clippy discovered warnings in the codebase" --body-file clippy.log
+          fi
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
It means, this GitHub repo must have a personal access token `GITHUB_TOKEN` configured for that. I do not have that privilege.

Reply:
Sorry for not updating this sooner - I was communicating with my team and I think we are going to back off the plan to create an issue for clippy warnings. There is a org-wide movement to move away from using tokens/secrets of any kind and introducing another one right now is probably not the best idea.

Reply:
In terms of running clippy periodically, we can split it into a separate workflow if you think that would be best. I was planning on just having the entire CI workflow run every Sunday at midnight in addition to on PR creation / pushes to main just to minimize the number of workflows we have running. But, if we think it's best to just split it up, it's not hard to do and I can close this PR and just create a separate clippy workflow.